File: src/SkyBlock/enchants/touch/BaseTouchEnchant.php
<?php


namespace SkyBlock\enchants\touch;


use pocketmine\block\Block;
use pocketmine\event\player\PlayerInteractEvent;
use pocketmine\player\Player;
use SkyBlock\enchants\BaseEnchantment;

abstract class BaseTouchEnchant extends BaseEnchantment {
    abstract public function isApplicableTo(Player $holder, int $enchlevel = 0, Block $block = null) : bool;

    /**
     * @param Player              $player
     * @param PlayerInteractEvent $ev
     * @param int                 $enchantmentlevel
     */
    abstract public function onActivation(Player $player, PlayerInteractEvent $ev, int $enchantmentlevel) : void;
}

File: SkyblockCore/src/main/java/dev/skyblock/enchants/BaseTouchEnchant.java
package dev.skyblock.enchants;
import org.bukkit.entity.Player;
import org.bukkit.event.player.PlayerInteractEvent;
public abstract class BaseTouchEnchant extends BaseEnchant {
    public BaseTouchEnchant(int id, String name, int maxLevel) { super(id, name, maxLevel); }
    public abstract void onActivation(Player player, PlayerInteractEvent ev, int level);
}
